src/ImageSharp/Image/PixelAccessor{TColor}.cs

```diff
@@ -186,8 +186,7 @@ public void CopyFrom(PixelArea<TColor> area, int targetY, int targetX = 0)
             int width = Math.Min(area.Width, this.Width - targetX);
             int height = Math.Min(area.Height, this.Height - targetY);
 
-            this.CheckDimensions(width, height);
-
+            // this.CheckDimensions(width, height); TODO: Why was width == 0 or height == 0 considered a problem? Copy implementations do not fail on this (just do nothing)!
```

Comment:
How about explicit method calls instead of a switch then? More boilerplate for us but should yield much better performance.